### PR TITLE
replace calls to addr_to_string with LogAddr

### PR DESCRIPTION
**Pull request by lij-oci**

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -2796,20 +2792,15 @@ Spdp::SpdpTransport::choose_send_socket(const ACE_INET_Addr& addr) const
 void
 Spdp::SpdpTransport::send(const ACE_INET_Addr& addr)
 {
-  ssize_t res;
-
   const ACE_SOCK_Dgram& socket = choose_send_socket(addr);
-  res = socket.send(wbuff_.rd_ptr(), wbuff_.length(), addr);
-
+  ssize_t res = socket.send(wbuff_.rd_ptr(), wbuff_.length(), addr);
```
Comment (simpsont-oci) [inline dds/DCPS/RTPS/Spdp.cpp:2796]:
```suggestion
  const ssize_t res = socket.send(wbuff_.rd_ptr(), wbuff_.length(), addr);
```

File: dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp [hunk 2]
```diff
@@ -648,12 +646,10 @@ namespace {
     if (result < 0) {
       const int err = errno;
       if (err != ENETUNREACH || !network_is_unreachable) {
-        ACE_TCHAR addr_buff[DCPS::AddrToStringSize] = {};
-        addr.addr_to_string(addr_buff, DCPS::AddrToStringSize);
         errno = err;
         const ACE_Log_Priority prio = shouldWarn(errno) ? LM_WARNING : LM_ERROR;
         ACE_ERROR((prio, "(%P|%t) RtpsUdpTransport.cpp send_single_i() - "
-                   "destination %s failed %p\n", addr_buff, ACE_TEXT("send")));
+                   "destination %C failed %p\n", DCPS::LogAddr(addr).c_str(), ACE_TEXT("send")));
```
Comment (simpsont-oci) [inline dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp:652]: You're missing a few migrations from to_dds_string() (lines 841, 846, 875). I'd comment on them directly, but GitHub won't let me make a comment on a line that's not part of the diff.

File: dds/DCPS/SafetyProfileStreams.cpp [hunk 3]
```diff
@@ -7,6 +7,7 @@
 #include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
 
 #include "SafetyProfileStreams.h"
+#include "LogAddr.h"
```
Comment (simpsont-oci) [inline dds/DCPS/SafetyProfileStreams.cpp:10]:
```suggestion
```

**Review by simpsont-oci — CHANGES_REQUESTED** [review]
Overall looks good, with a few minor tweaks.

File: dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp (hunk 2 above)
Comment (lij-oci) [reply dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp:652]: Good catch! Thanks.

**Review by simpsont-oci — APPROVED** [review]

**Pull request merged**